pkg/operator/ceph/cluster/watcher_test.go

```diff
@@ -32,6 +32,43 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
+func TestOnK8sNode(t *testing.T) {
+	service := &corev1.Service{}
+	ns := "rook-ceph"
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+	s.AddKnownTypes(cephv1.SchemeGroupVersion, &cephv1.CephCluster{})
+
+	cephCluster := &cephv1.CephCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ns,
+			Namespace: ns,
+		},
+		Status: cephv1.ClusterStatus{
+			Phase: "",
+		},
+	}
+
+	object := []runtime.Object{
+		cephCluster,
+	}
+
+	// Create a fake client to mock API calls.
+	cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(object...).Build()
```

Comment:
you can move the code snippet that creates the fake client as a separate method. Then we can use the same method in both the tests in this file.